src/textual/_compositor.py

```diff
@@ -604,19 +604,28 @@ def _get_renders(
         # up to this point.
         _rich_traceback_guard = True
 
+        def is_visible(widget: Widget) -> bool:
```

Comment:
This is a fairly tight loop.

Humor me and unroll the code in is_visible. Essentially replace the properties with immediate code, so it makes no implicity function calls.